### ISSUE TITLE
externally accessible functions

### ISSUE THREAD

**Issue by MalteLond** [lorenzha/hdcd#47]
Which functions should be externally accessible. I propose, similar to glmnet, two functions:

- hdcd, similar to currently BinarySegmentation
- cv.hdcd, similar to CrossValidation

I do not like the idea of having one function that does cv if one parameter is missing or similar. Maybe throw a warning, choose lambda according to asymptotic theory, delta = 0.1 and return a list of cps corresponding to different gammas if no params are given. Choose a default method (e.g. glasso) and a default optimizer.

**Comment by lorenzha** [comment]
I agree. Also adapting the structure from glmnet is generally a good idea.

**Comment by MalteLond** [comment]
Possibly: Change name away from hdcd? E.g. FastChangepointDetection.

**Comment by lorenzha** [comment]
Also generic functions like e.g. plot and print...
